MediaBrowser.Providers/MediaInfo/SubtitleResolver.cs

```diff
@@ -131,51 +135,55 @@ public SubtitleResolver(ILocalizationManager localization)
                         Path = fullName
                     };
                 }
-                else if (fileNameWithoutExtension.Length > videoFileNameWithoutExtension.Length
-                         && fileNameWithoutExtension[videoFileNameWithoutExtension.Length] == '.'
-                         && fileNameWithoutExtension.StartsWith(videoFileNameWithoutExtension, StringComparison.OrdinalIgnoreCase))
+                else if (_compareInfo.IsPrefix(fileNameWithoutExtension, videoFileNameWithoutExtension, CompareOptions, out int matchLength)
+                         && fileNameWithoutExtension[matchLength] == '.')
                 {
                     var isForced = fullName.Contains(".forced.", StringComparison.OrdinalIgnoreCase)
                                    || fullName.Contains(".foreign.", StringComparison.OrdinalIgnoreCase);
 
                     var isDefault = fullName.Contains(".default.", StringComparison.OrdinalIgnoreCase);
 
                     // Support xbmc naming conventions - 300.spanish.srt
+                    string? language = null;
+                    string? title = null;
                     var languageSpan = fileNameWithoutExtension;
                     while (languageSpan.Length > 0)
                     {
                         var lastDot = languageSpan.LastIndexOf('.');
-                        if (lastDot < videoFileNameWithoutExtension.Length)
+                        if (lastDot < matchLength)
                         {
-                            languageSpan = ReadOnlySpan<char>.Empty;
                             break;
                         }
 
                         var currentSlice = languageSpan[lastDot..];
+                        languageSpan = languageSpan[..lastDot];
+
                         if (currentSlice.Equals(".default", StringComparison.OrdinalIgnoreCase)
                             || currentSlice.Equals(".forced", StringComparison.OrdinalIgnoreCase)
                             || currentSlice.Equals(".foreign", StringComparison.OrdinalIgnoreCase))
                         {
-                            languageSpan = languageSpan[..lastDot];
                             continue;
                         }
 
-                        languageSpan = languageSpan[(lastDot + 1)..];
-                        break;
```

Comment:
You no longer break?

Reply:
Intentional - the old way was pulling the last non-default/forced/foreign segment between "." delimiters as the language and ignoring anything else.

Switching to using the actual title field when it's not a language means collecting all the segments that don't parse as a language or special text, not just the last one. The `lastDot < matchLength` check will keep it from pulling the end of the movie name if it's "."-separated.